Microsoft.HealthVault.Fhir/FhirExtensions/FhirDateTimeExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using Hl7.Fhir.Model;
+
+namespace Microsoft.HealthVault.Fhir.FhirExtensions
+{
+    public static class FhirDateTimeExtensions
+    {
+        public enum FhirDateTimePrecision
+        {
+            Year = 4,       //2017
+            Month = 7,      //2017-22
+            Day = 10,       //2017-22-21
```

Comment:
FhirDateTime expects in the format 2017-07-28
As per regular expression of FhirDateTime, month and date should have 2 digits 
( https://www.hl7.org/fhir/datatypes.html#dateTime )